configd/src/apps/sentinel/report-connectivity.h

```diff
@@ -0,0 +1,63 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include <vespa/fnet/frt/rpcrequest.h>
+#include <vespa/fnet/frt/supervisor.h>
+#include <vespa/config-model.h>
+#include <vespa/config/helper/configfetcher.h>
+#include "peer-check.h"
+#include "status-callback.h"
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <vector>
+
+namespace config::sentinel {
+
+class ReportConnectivity;
+
+struct SinglePing : StatusCallback {
+    ReportConnectivity& parent;
+    std::string peerName;
+    int peerPort;
+    std::string status;
+    std::unique_ptr<PeerCheck> check;
+
+    SinglePing(ReportConnectivity& owner, const std::string &hostname, int port)
+      : parent(owner),
+        peerName(hostname),
+        peerPort(port),
+        status("unknown"),
+        check(nullptr)
+    {}
+
+    SinglePing(SinglePing &&) = default;
+    SinglePing(const SinglePing &) = default;
+
+    virtual ~SinglePing();
+    void startCheck(FRT_Supervisor &orb);
+    void returnStatus(bool ok) override;
```

Comment:
consider extending callback with hostname to avoid needing multiple ping classes